File: 3601-find-the-k-th-character-in-string-game-ii/3601-find-the-k-th-character-in-string-game-ii.java
class Solution {
    public char kthCharacter(long k, int[] operations) {
    long  ops = 0;
    while(k > 1){
        int p = (int)Math.ceil(Math.log(k)/Math.log(2));
        k -= Math.pow(2, p - 1);
        ops += operations[p - 1]; 
    }
    return (char)('a' + (ops % 26));
    }
}

File: 3601-find-the-k-th-character-in-string-game-ii/3601-find-the-k-th-character-in-string-game-ii.cpp
class Solution {
public:
    char kthCharacter(long long k, vector<int>& operations) {
      
        int noOfOps = 0;
        while(k > 1){
            int p = ceil(log2(k));
            k -= pow(2, p - 1);
            noOfOps += operations[p - 1];
        }        

        return (char)('a' + (noOfOps % 26));
    }
};